### PR TITLE
Drop Python 3.2 support, refactor Travis config, add flake8

### PR DESCRIPTION
**Pull request by bittner**
This PR:

1. **Drops Python 3.2 support** since many packages that this package and CI setup depends on have already done so (it's starting to get a dependency maintenance nightmare)
1. **Refactors Travis CI configuration** to make it more stable and nicer to look at (table view with Python versions)
1. **Adds flake8 checks** to both Tox and Travis, and addresses all complaints (a massive amount of files, but almost all only indentation and whitespace issues)
1. Python 3.6 and Django 1.10 + 1.11 **support is prepared** in the Tox, Travis and setup configuration, but requires extending the included test project.

### CODE REVIEW

File: analytical/tests/test_tag_uservoice.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def test_override_options(self):
         r = UserVoiceNode().render(Context(data))
         self.assertIn("""UserVoice.push(['set', {"key1": "val2"}]);""", r)
 
-    def test_auto_trigger(self):
+    def test_auto_trigger_default(self):
```
Comment (bittner) [inline analytical/tests/test_tag_uservoice.py]: @jcassee This is what I found (and hopefully fixed correctly) by help of `flake8`.

File: analytical/tests/test_utils.py [hunk 2]
```diff
@@ -30,11 +30,11 @@ def test_get_required_setting(self):
         # available in python >= 3.2
         if hasattr(self, 'assertRaisesRegex'):
             with self.assertRaisesRegex(AnalyticalException, "^USER_ID setting is not set$"):
-                user_id = get_required_setting("USER_ID", "\d+", "invalid USER_ID")
+                get_required_setting("USER_ID", "\d+", "invalid USER_ID")
         # available in python >= 2.7, deprecated in 3.2
         elif hasattr(self, 'assertRaisesRegexp'):
             with self.assertRaisesRegexp(AnalyticalException, "^USER_ID setting is not set$"):
-                user_id = get_required_setting("USER_ID", "\d+", "invalid USER_ID")
+                get_required_setting("USER_ID", "\d+", "invalid USER_ID")
```
Comment (bittner) [inline analytical/tests/test_utils.py]: I removed the two assignments, `flake8` complained about that too. Should be fine, I guess.

File: tox.ini [hunk 3]
```diff
@@ -1,19 +1,39 @@
...
     coverage run setup.py test
     sh -c 'coveralls | true'
 deps =
-    coverage==3.7.1
+    coverage
     coveralls
     django17: Django>=1.7,<1.8
     django18: Django>=1.8,<1.9
     django19: Django>=1.9,<1.10
-    virtualenv<14.0.0
```
Comment (bittner) [inline tox.ini]: Pinning `coverage` and `virtualenv` was necessary only for Python 3.2.

**Comment by jcassee** [comment]
Merge away!

**Comment by bittner** [comment]
I'm really not happy with the huge `.travis-ci.yml` file we have now, but that's probably as good as it gets.

See https://github.com/tox-dev/tox-travis/issues/77#issuecomment-336804896 for reference.

**Pull request merged**